data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/consumer/KafkaCustomConsumer.java

```diff
@@ -496,7 +497,7 @@ private <T> void iterateRecordPartitions(ConsumerRecords<String, T> records, fin
                 if (schema == MessageFormat.BYTES) {
                     InputStream inputStream = new ByteArrayInputStream((byte[])consumerRecord.value());
                     if(byteDecoder != null) {
-                        byteDecoder.parse(inputStream, (record) -> {
+                        byteDecoder.parse(inputStream, Instant.ofEpochMilli(consumerRecord.timestamp()), (record) -> {
```

Comment:
Can `consumerRecord.timestamp()` ever be `null`? If so, this will throw an exception.

Reply:
I do not think that is possible.